pykafka/balancedconsumer.py

```diff
@@ -535,7 +566,24 @@ def _rebalance(self):
 
                     # Only re-create internal consumer if something changed.
                     if new_partitions != self._partitions:
-                        self._setup_internal_consumer(list(new_partitions))
+                        cns = self._get_internal_consumer(list(new_partitions))
+                        if self._consumer is not None and \
+                                self._partitions_revoked_callback is not None:
+                            self._partitions_revoked_callback(
+                                self, _get_held_offsets_for_partitions(
+                                    current_zk_parts - new_partitions, self._consumer))
+                        if self._partitions_assigned_callback is not None:
+                            new_offsets = self._partitions_assigned_callback(
+                                self,
+                                _get_held_offsets_for_partitions(new_partitions, cns))
```

Comment:
Oh wait - the docstring says it is all partitions. I'm wondering which is more convenient though? If you really wanted all partitions you could get those off the `self` that also gets passed, but figuring out which ones are new is more tricky?